### PR TITLE
feat: select support status

### PR DESCRIPTION
**Pull request by MadCcc**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/33854
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |       Select support custom status    |
| 🇨🇳 Chinese |       Select 组件支持自定义状态    |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Review by zombieJ — APPROVED** [review]

File: components/select/utils/iconUtil.tsx [hunk 1]
```diff
@@ -13,35 +16,49 @@ export default function getIcons({
...
     mergedClearIcon = <CloseCircleFilled />;
   }
 
+  // Validation Feedback Icon
+  const getSuffixIconNode = (arrowIcon?: ReactNode) => (
+    <>
+      {showArrow !== false && arrowIcon}
+      {hasFeedback && getFeedbackIcon(prefixCls, status)}
+    </>
+  );
```
Comment (afc163) [inline components/select/utils/iconUtil.tsx:47]: getSuffixIconNode 单独写一个 `<SuffixIcon showArrow hasFeedback prefixCls status>{icon}</SuffixIcon>` 组件如何？

Comment (afc163) [reply components/select/utils/iconUtil.tsx:47]: getFeedbackIcon 也类似。

Comment (afc163) [reply components/select/utils/iconUtil.tsx:47]: 用函数 + 闭包感觉不那么 FP。

**Comment by sly9502** [comment]
原来后缀icon还支持函数，可以用 open 等 props

**Pull request merged**